### PR TITLE
[Compiler-v2] Check `dummy_field` in a struct

### LINKED ISSUE #13594: [Bug][Compiler-v2] handling dummy_field in V2 is problematic
# 🐛 Bug

When compiling the following code using V2:

```
module 0x42::test {
    
    struct R has store, key {
        dummy_field: bool,
    }
    
    public entry fun test(addr: address) acquires R {
        let R {  } = move_from<R>(addr);
    }
    
}
```

V2 will pass while V1 will generate the error:

```error[E04016]: too few arguments
  ┌─ test.move:9:13
  │
9 │         let R {  } = move_from<R>(arg0);
  │             ^^^^^^ Missing binding for field 'dummy_field' in '0x42::test::R'`.
 ```

### PR DESCRIPTION
**Pull request by rahxephon89**
## Description

This PR adds a new bool field `is_empty_struct` to `StructEntry` to indicate whether the struct does not have any fields. With this field, 

- if a user defines a struct with the field `dummy_field`, packing and unpacking without any fields will raise an error;
- accessing `dummy_field` from an empty struct will raise an error.

close #13594 

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

1) Existing tests pass;
2) Add a new test `dummy_field.move`. 

## Key Areas to Review

Whether the added logic will break how struct fields are generally checked.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by wrwg — APPROVED** [review]

File: third_party/move/move-compiler-v2/tests/checking/typing/dummy_field.move [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        let T {  } = move_from<T>(addr);
+    }
+
+    public entry fun test4(s: &signer) {
+        let r = T {};
+        move_to<T>(s, r);
+    }
+
+    fun test_access_implicit_dummy(): bool {
+        let t = T {
```
[the remark this answers is not in the document]
Comment (rahxephon89) [reply third_party/move/move-compiler-v2/tests/checking/typing/dummy_field.move]: Added the test

File: third_party/move/move-compiler-v2/tests/checking/typing/dummy_field.move [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    public entry fun test(addr: address) acquires R {
+        let R {  } = move_from<R>(addr);
+    }
+
+    public entry fun test2(s: &signer) {
+        let r = R {};
+        move_to<R>(s, r);
+    }
+
+    fun test_access_explicit_dummy(): bool {
```
[the remark this answers is not in the document]
Comment (rahxephon89) [reply third_party/move/move-compiler-v2/tests/checking/typing/dummy_field.move]: Added the test and separate positive from negative ones.

**Review by brmataptos — APPROVED** [review]

File: third_party/move/move-model/src/builder/module_builder.rs [hunk 3]
```diff
@@ -1185,6 +1187,11 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
             .get_mut(&qsym)
             .expect("struct invalid")
             .fields = fields;
+        self.parent
```
[the remark this answers is not in the document]
Comment (rahxephon89) [reply third_party/move/move-model/src/builder/module_builder.rs]: done

**Review by vineethk — APPROVED** [review]
LGTM.

**Review by fEst1ck — APPROVED** [review]
LGTM

**Review by brmataptos — APPROVED** [review]

**Pull request merged**